python/lib/dcos/dcos/mesos.py

```diff
@@ -493,17 +493,33 @@ def tasks(self, fltr=None, completed=False, all_=False):
 
         return tasks
 
-    def get_container_id(self, task_obj):
-        """Returns the container ID for a task.
+    def get_container_id(self, task_id):
+        """Returns the container ID for a task ID matching `task_id`
 
-        :param task__obj: The task which will be mapped to container ID
-        :type task_obj: Task()
-        :returns: The container ID associated with 'task_obj'
+        :param task_id: The task ID which will be mapped to container ID
+        :type task_id: str
+        :returns: The container ID associated with 'task_id'
         :rtype: str
         """
 
-        def _get_container_status(task_obj):
-            task = task_obj.dict()
+        def _get_task(task_id):
+            candidates = []
+            if 'frameworks' in self.state():
+                for framework in self.state()['frameworks']:
+                    if 'tasks' in framework:
+                        for task in framework['tasks']:
+                            if 'id' in task:
+                                if task['id'].startswith(task_id):
+                                    candidates.append(task)
```

Comment:
There are a lot of indentation levels, if conditions before loops could be removed by using the `dict.get('key', default)` syntax:
``` python
for framework in self.state().get('frameworks', []):
    for task in framework.get('tasks', []):
        # ...
```